yt_dlp/extractor/rcs.py

```diff
@@ -90,112 +72,91 @@ class RCSBaseIE(InfoExtractor):
+                    'bitrate': source.get('bitrate')})
 
-        # replace host
-        for t in src:
+        for source in sources:
             for s, r in self._ALL_REPLACE.items():
-                src[t] = src[t].replace(s, r)
-            for s, r in self._MP4_REPLACE.items():
-                src[t] = src[t].replace(s, r)
+                source['url'] = source['url'].replace(s, r)
 
-        # switch cdn
-        if 'mp4' in src and 'm3u8' in src:
-            if ('-lh.akamaihd' not in src.get('m3u8')
-                    and 'akamai' in src.get('mp4')):
-                if 'm3u8' in src:
-                    matches = re.search(r'(?:https*:)?\/\/(?P<host>.*)\.net\/i(?P<path>.*)$', src.get('m3u8'))
-                    src['m3u8'] = 'https://vod.rcsobjects.it/hls/%s%s' % (
-                        self._MIGRATION_MAP[matches.group('host')],
-                        matches.group('path').replace(
-                            '///', '/').replace(
-                            '//', '/').replace(
-                            '.csmil', '.urlset'
-                        )
-                    )
-                if 'mp4' in src:
-                    matches = re.search(r'(?:https*:)?\/\/(?P<host>.*)\.net\/i(?P<path>.*)$', src.get('mp4'))
-                    if matches:
-                        if matches.group('host') in self._MIGRATION_MEDIA:
-                            vh_stream = 'https://media2.corriereobjects.it'
-                            if src.get('mp4').find('fcs.quotidiani_!'):
-                                vh_stream = 'https://media2-it.corriereobjects.it'
-                            src['mp4'] = '%s%s' % (
-                                vh_stream,
-                                matches.group('path').replace(
-                                    '///', '/').replace(
-                                    '//', '/').replace(
-                                    '/fcs.quotidiani/mediacenter', '').replace(
-                                    '/fcs.quotidiani_!/mediacenter', '').replace(
-                                    'corriere/content/mediacenter/', '').replace(
-                                    'gazzetta/content/mediacenter/', '')
-                            )
-                        else:
-                            src['mp4'] = 'https://vod.rcsobjects.it/%s%s' % (
-                                self._MIGRATION_MAP[matches.group('host')],
-                                matches.group('path').replace('///', '/').replace('//', '/')
-                            )
+            if source['type'] == 'm3u8' and '-lh.akamaihd' in source['url']:
+                matches = re.search(r'(?:https*:)?\/\/(?P<host>.*)\.net\/i(?P<path>.*)$', source['url'])
+                source['url'] = 'https://vod.rcsobjects.it/hls/%s%s' % (
+                    self._MIGRATION_MAP[matches.group('host')],
+                    matches.group('path').replace('///', '/').replace('//', '/').replace('.csmil', '.urlset')
+                )
```

Comment:
this can be simplified. But since it's legacy code, I wont insist

Reply:
yup, I know.

It's because the code was written as a 1:1 translation of the js player code, so it was easier to spot any changes in the original js code and implement it in the py code of the extractor. After the initial writing of the extractor code the js player code is still the same, so I guess we can simplify it.

Reply:
After some tests, I think I can even remove it, because it seems like legacy code that is not used anymore. All the urls I tested didn't trigger the if condition of this piece of code.

It looks something they implemented when they were switching CDN and it was necessary to correct the manifest url but now it's no longer necessary.